extra/innochecksum.cc

```diff
@@ -1846,13 +1913,19 @@ int main(
 			    && !is_page_free(xdes, physical_page_size, cur_page_num)
 			    && (exit_status = verify_checksum(
 						buf, is_encrypted,
-						&mismatch_count, flags))) {
+						&mismatch_count,
+						tablespace_flags))) {
 				goto my_exit;
 			}
 
-			if ((exit_status = rewrite_checksum(
-						filename, fil_in, buf,
-						&pos, is_encrypted, flags))) {
+			/* Skip rewrite_checksum for doublewrite buffer pages
+			when skip_freed_pages is enabled */
+			if ((!skip_freed_pages || !is_system_tablespace ||
+			     !is_page_doublewritebuffer()) &&
+			    (exit_status = rewrite_checksum(
```

Comment:
Is there any valid reason to ever rewrite the checksum on doublewrite buffer pages? Can we simplify the condition?